File: src/01-easy/mime-type/solution.py
n = int(input())
q = int(input())
m = {}
for _ in range(n):
    parts = input().split()
    m[parts[0].lower()] = parts[1]
for _ in range(q):
    f = input()
    dot = f.rfind('.')
    if dot == -1:
        print('UNKNOWN')
    else:
        print(m.get(f[dot+1:].lower(), 'UNKNOWN'))

File: src/01-easy/mime-type/solution.rb
n = gets.to_i
q = gets.to_i
m = {}
n.times { p = gets.split; m[p[0].downcase] = p[1] }
q.times {
  f = gets.chomp
  dot = f.rindex('.')
  puts dot ? (m[f[dot+1..].downcase] || 'UNKNOWN') : 'UNKNOWN'
}
